novatel_gps_driver/src/nodelets/novatel_gps_nodelet.cpp

```diff
@@ -356,6 +369,51 @@ namespace novatel_gps_driver
     }
 
     /**
+        * SystemAlert to shutdown the sensor node
+        */
+      void alertCallback(const cav_msgs::SystemAlertConstPtr &msg)
+       {
+         if( msg->type==cav_msgs::SystemAlert::FATAL || msg->type==cav_msgs::SystemAlert::SHUTDOWN)
+           {
+            ros::shutdown();
+           }
+       }
+
+      /**
+          * DriverStatus function definition various novatel driver status
+          */
+      void publish_status(const ros::TimerEvent&)
+      {  // Set driver type
+          status_.gps=true;
+          status_.imu=true;
+
+          time_difference=ros::Time::now()-last_update_time_;
+
+          if (last_update_time_.isZero() || ((time_difference>ros::Duration(2.0)) && (status_gps==cav_msgs::DriverStatus::OPERATIONAL)))
```

Comment:
I think 2 seconds here is much longer than we need. Half a second should be plenty to detected a timeout. 

Additionally, please add a log for status changes.